src/core/core/rpc_message.cpp

```diff
@@ -216,11 +222,26 @@ DSN_API void dsn_msg_get_options(
     opts->gpid = hdr->gpid;
 }
 
+DSN_API dsn_msg_header_type dsn_msg_get_header_type(
+    dsn_message_t msg
+    )
+{
+    dsn::header_type& type = ((::dsn::message_ex*)msg)->header->hdr_type;
+    if (type == dsn::header_type::hdr_dsn_default)
```

Comment:
let's simplify this by e.g., letting header_type::hdr_dsn_default == DHT_DEFAULT